compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+                                   ds_mesh.areaCell * 0.5,
+                                   ds_mesh.areaCell)
+
+    # Needed of reducing the edge lengths by half along the N/S boundary.
+    # Values returned by `np.unique` will be sorted.
+    unique_ys_edge = np.unique(ds_mesh.yEdge.data)
+
+    # Boolean mask for edge indices on the N/S boundary of the mesh
+    mask = ((ds_mesh.yEdge == unique_ys_edge[0]) |
+            (ds_mesh.yEdge == unique_ys_edge[-1]))
+    # WHL: zero out the edges on the boundary
+    #      (not necessary because velocity will also be zero)
+    ds_mesh['dvEdge'] = xr.where(mask, 0.0, ds_mesh.dvEdge)
+
+    # Boolean mask for the indexed of edges between N/S boundary cells
+    mask = ((ds_mesh.yEdge == unique_ys_edge[1]) |
+            (ds_mesh.yEdge == unique_ys_edge[-2]))
+    # cut length in half for edges between boundary cells
+    ds_mesh['dvEdge'] = xr.where(mask, ds_mesh.dvEdge * 0.5, ds_mesh.dvEdge)
+
+    return ds_mesh
+
+
+def _mismipplus_bed(x, y):
+    """
+    Equations 1-4 from Asay-Davis et al. (2016).
+
+    Parameters
+    ----------
+    x : xarray.DataArray
+        x cell coordinates
+
+    y : xarray.DataArray
+        y cell coordinates
+    """
+    x = x / 1.e3      # m to km
+    y = y / 1.e3      # m to km
+    B0 = -150.        # m
+    B2 = -728.8       # m
+    B4 = 343.91       # m
+    B6 = -50.57       # m
+    x_bar = 300.      # km
+    x_tilde = x / x_bar
+    dc = 500.         # m
+    fc = 4.           # km
+    wc = 24.          # km
+    Ly = 80.          # km
+    Bmax = -720.      # m
+
+    B_x = B0 + B2 * x_tilde**2 + B4 * x_tilde**4 + B6 * x_tilde**6
+    B_y = dc / (1 + np.exp(-2 * (y - Ly / 2 - wc) / fc)) +\
+        dc / (1 + np.exp(2 * (y - Ly / 2 + wc) / fc))
+
+    Bsum = B_x + B_y
+
+    z_b = np.maximum(Bsum, Bmax)  # Eqn 1. (Asay-Davis et al. 2016)
+
+    return z_b
+
+
+def _setup_MISMPPlus_IC(config, logger, filename):
+    """
+    Add the inital condition for the MISMIP+ spinup to the given MPAS mesh file
+
+    Parameters
+    ----------
+    config : comass.config.CompassConfigParser
+        Configuration options for this test case
+
+    logger : logging.Logger
+        A logger for output from the step
+
+    filename : str
+        NetCDF file to place the MISMIP+ ICs in
+
+    """
+
+    # Hard code some parameters from Table 1. of Asay-Davis et al. 2016
+    accum = 0.3       # m^{-1}
+    C = 3.160e6       # Pa m^{-1/3} s^{1/3}
+    rhoi = 918        # kg m^{-3}
+    spy = 31556926    # s a^{-1}
+    xcalve = 640.e3   # m
+
+    # Read parameters from the .cfg file
+    section = config['mesh']
+    nVertLevels = section.getint('levels')
+    init_thickness = section.getfloat('init_thickness')
+
+    # open the file
+    src = xr.open_dataset(filename)
+
+    # Use `.loc[:]` for setting the initial conditions  since we are setting
+    # the fields with scalar values. This ensures values are properly broadcast
```

Comment:
I'm not familiar with the `.loc` attribute, but I'm confused by this comment.  It looks the fields below are spatially varying so I don't follow the comment about scalar values here.

Reply:
Yeah I need to update that comment to make it more accurate. I'm using it here to ensure broadcasting and alignment work correctly in variable assignment. When a scalar value is being assigned to a field `.loc[:]` will ensure that the scalar value is broadcast against all the dimensions of the field and that the broadcasting is aligned properly . For example, if we left off the `.loc[:]` and did something like `src.example_field = 910.` then the resulting example field would have all it's dimensions (e.g. `Time`, `nCells`, `nVertInterfaces`) dropped and just be a scalar value. 

But the need for `.loc[:]` in field assignment is more general. For example the `mask` variable in the `_setup_MISMPPlus_IC` function  only contains dimensions `nCells` (since it was created from `src.yCell` which itself only contains dimensions `nCells`). So when we use the `mask` variable to assign fields (e.g. `dirichletVelocityMask`) the inclusion of `.loc[:]` ensure that the value we are assigning are broadcast and aligned against all the dimensions of the field on the right hand side (`Time`, `nCells`, `nVertInterfaces`). 

So more generally, `.loc[:]` is needed anytime any time a variable is being used for assignment which does not contain all the dimensions of the field you're trying to set. Therefore, `.loc[:]` is needed in all the IC prescriptions because all variables in the mesh contain a `Time` dimension, but none of the fields used to set the IC contain that `Time` dimension. 

Hopefully that description is clear. I'll update the comment here to reflect the details as elaborated here more accurately.